l5kit/l5kit/tests/simulation/e2e_test.py

```diff
@@ -0,0 +1,122 @@
+    assert agg["distance_ref_trajectory_validator"].item() == 0
+    assert agg["collision_front_validator"].item() == 0
+    assert agg["collision_rear_validator"].item() == 0
+    assert agg["collision_side_validator"].item() == 0
+
+
+@pytest.mark.parametrize("advance_x", [2.0, 0.0, 5.0])
+def test_e2e_ego(ego_cat_dataset: EgoDataset, simulation_evaluator: ClosedLoopEvaluator,
+                 advance_x: float) -> None:
+    sim_cfg = SimulationConfig(use_ego_gt=False, use_agents_gt=True, disable_new_agents=False,
+                               distance_th_far=1, distance_th_close=0, num_simulation_steps=20)
+
+    ego_model = MockModel(advance_x=advance_x)
+
+    sim_loop = ClosedLoopSimulator(sim_cfg, ego_cat_dataset, torch.device("cpu"),
+                                   model_ego=ego_model)
+
+    sim_out = sim_loop.unroll(list(range(len(ego_cat_dataset.dataset.scenes))))
+    simulation_evaluator.evaluate(sim_out)
+    agg = ValidationCountingAggregator().aggregate(simulation_evaluator.validation_results())
+
+    keys = ["displacement_error_l2_validator", "distance_ref_trajectory_validator",
+            "collision_front_validator", "collision_front_validator", "collision_side_validator",
+            "collision_rear_validator"]
+
+    if advance_x == 0.0:  # bumps by rear car
+        key_to_trigger = "collision_rear_validator"
+    elif advance_x == 2.0:  # too fast
+        key_to_trigger = "distance_ref_trajectory_validator"
+    elif advance_x == 5.0:  # too fast bumps into leading car
+        key_to_trigger = "collision_front_validator"
+    else:
+        raise ValueError(f"advance_x {advance_x} not valid")
+
+    for k in keys:
+        if k == key_to_trigger:
+            assert agg[k].item() == 4
+        else:
+            assert agg[k].item() == 0
+
+
+@pytest.mark.parametrize("advance_x", [0.0, 2.0])
+def test_e2e_agents(ego_cat_dataset: EgoDataset, simulation_evaluator: ClosedLoopEvaluator,
+                    advance_x: float) -> None:
+    sim_cfg = SimulationConfig(use_ego_gt=True, use_agents_gt=False, disable_new_agents=False,
+                               distance_th_far=60, distance_th_close=30, num_simulation_steps=20,
+                               start_frame_index=10)  # start from 0 as leading is rotated at 0
+
+    agents_model = MockModel(advance_x=advance_x)
+
+    sim_loop = ClosedLoopSimulator(sim_cfg, ego_cat_dataset, torch.device("cpu"),
+                                   model_agents=agents_model)
+
+    sim_out = sim_loop.unroll(list(range(len(ego_cat_dataset.dataset.scenes))))
+    simulation_evaluator.evaluate(sim_out)
+    agg = ValidationCountingAggregator().aggregate(simulation_evaluator.validation_results())
+
+    keys = ["displacement_error_l2_validator", "distance_ref_trajectory_validator",
+            "collision_front_validator", "collision_front_validator", "collision_side_validator",
+            "collision_rear_validator"]
```

Comment:
I think it would be better to get these names directly from the evaluation plan, because if we change the other code to add more validators it will break here.